torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -157,16 +157,59 @@ void RRefContext::delUser(
     const worker_id_t owner,
     const RRefId& rrefId,
     const ForkId& forkId) {
-  std::lock_guard<std::mutex> lock(destroyedMutex_);
-  if (!destroyed_) {
-    auto fm = agent_->send(
-        agent_->getWorkerInfo(owner),
-        RRefUserDelete(rrefId, forkId).toMessage());
+  {
+    std::lock_guard<std::mutex> lock(destroyedMutex_);
+    if (!destroyed_) {
+      auto fm = agent_->send(
+          agent_->getWorkerInfo(owner),
+          RRefUserDelete(rrefId, forkId).toMessage());
+
+      fm->addCallback([](const Message& /* unused */,
+                         const c10::optional<utils::FutureError>& futErr) {
+        handleException(futErr);
+      });
+    }
+  }
 
-    fm->addCallback([](const Message& /* unused */,
-                       const c10::optional<utils::FutureError>& futErr) {
-      RRefContext::handleException(futErr);
-    });
+  std::lock_guard<std::mutex> lock(mutex_);
+  confirmedUsers_.erase(forkId);
+}
+
+void RRefContext::delAllUsers(std::chrono::milliseconds timeoutMillis) {
+  // First, wait for all pending UserRRefs to be confirmed,
+  // one kind is pendingUsers_, which are shared from Owner,
+  // the other kind pendingChildren_, which are shared from another User.
+  std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
looks like there is nothing that need to be shared across the three lock-unlock-lock blocks, we can use two lock_guard here to make the code more readable?

Reply:
sounds good. updating.